WMF Framework/NavigationBar.swift

```diff
@@ -170,6 +172,11 @@ public class NavigationBar: SetupView {
         _extendedViewPercentHidden = extendedViewPercentHidden
         setNeedsLayout()
         //print("nb: \(navigationBarPercentHidden) ev: \(extendedViewPercentHidden)")
+        if let extendedViewPercentHiddenForShowingTitle = self.extendedViewPercentHiddenForShowingTitle {
+            UIView.animate(withDuration: 0.2) {
+                self.delegate?.title = extendedViewPercentHidden >= extendedViewPercentHiddenForShowingTitle ? self.title : nil
```

Comment:
Could this go in the `changes` block below (which gets animated with same duration)? Would avoid potential situation where the two animation blocks could fight a bit...

Reply:
the title animation wouldn't be animated then, I can change it to use completion?

        let applyChanges = {
            let changes = {
                self.layoutIfNeeded()
                additionalAnimations?()
            }
            if animated {
                UIView.animate(withDuration: 0.2, animations: changes)
            } else {
                changes()
            }
        }
        if let extendedViewPercentHiddenForShowingTitle = self.extendedViewPercentHiddenForShowingTitle {
            UIView.animate(withDuration: 0.2, animations: {
                self.delegate?.title = extendedViewPercentHidden >= extendedViewPercentHiddenForShowingTitle ? self.title : nil
            }, completion: { (_) in
                applyChanges()
            })
        } else {
            applyChanges()
        }

Reply:
Ya that looks like that would keep the 2 animations from trying to happen simultaneously.

Can you check if the `self.layoutIfNeeded()` call in the `changes` block causes the `navigationBarPercentHidden` and `extendedViewPercentHidden` changes to not animate? At first glance it looks like only `additionalAnimations` changes would animate as a result...

Reply:
Ah sorry - in the case animated is true, it looks like the primary things being animated were the changes to `navigationBarPercentHidden` and `extendedViewPercentHidden`, and then there are `additionalAnimations`. 

But the call to `self.layoutIfNeeded()` looks like it could force the layout of the `navigationBarPercentHidden` and `extendedViewPercentHidden` changes instead of animating them.

Reply:
so are you saying that calling `self.layoutIfNeeded()` inside `UIView.animate` will not animate the layout of the `navigationBarPercentHidden` and `extendedViewPercentHidden` changes? 🤔

Reply:
ya i thought `layoutIfNeeded` forced immediate layout

Reply:
(Based on slack conversation should be good to go.)